packages/starlight/utils/translations-fs.ts

```diff
@@ -23,15 +28,18 @@ export function createTranslationSystemFromFs<T extends i18nSchemaOutput>(
 		// Load the user’s i18n directory
 		const files = fs.readdirSync(i18nDir, 'utf-8');
 		// Load the user’s i18n collection and ignore the error if it doesn’t exist.
-		userTranslations = Object.fromEntries(
-			files
-				.filter((file) => file.endsWith('.json'))
-				.map((file) => {
-					const id = file.slice(0, -5);
-					const data = JSON.parse(fs.readFileSync(new URL(file, i18nDir), 'utf-8'));
-					return [id, data] as const;
-				})
-		);
+		for (const file of files) {
+			const filePath = path.parse(file);
+			if (!contentCollectionFileExtensions.includes(filePath.ext)) continue;
+			const id = filePath.name;
+			const url = new URL(filePath.base, i18nDir);
+			const content = fs.readFileSync(new URL(file, i18nDir), 'utf-8');
+			const data =
+				filePath.ext === '.json'
+					? JSON.parse(content)
+					: yaml.load(content, { filename: fileURLToPath(url) });
```

Comment:
As YAML is a superset of JSON, we could technically have only used the YAML parser, although I decided against it for 2 reasons:

- We have a test erroring when parsing an empty JSON file, which would no longer have been an error if we used the YAML parser.
- For malformed JSON files, an YAMLException would be thrown, which I think could be confusing for users to mention YAML when they are using JSON files.